subsys/net/ip/dhcpv4.c

```diff
@@ -128,12 +128,10 @@ static inline bool dhcpv4_add_end(struct net_pkt *pkt)
 /* File is empty ATM */
 static inline bool dhcpv4_add_file(struct net_pkt *pkt)
 {
-	u8_t len = SIZE_OF_FILE;
+	const u16_t size = SIZE_OF_FILE;
 
-	while (len-- > 0) {
-		if (!net_pkt_append_u8(pkt, 0)) {
-			return false;
-		}
+	if (net_pkt_append_memset(pkt, size, 0, K_FOREVER) != size) {
```

Comment:
We are trying to migrate away from K_FOREVER timeout in #9118 so please use here a proper timeout value.